modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/index.d.ts

```diff
@@ -0,0 +1,213 @@
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+export function FrontendDataSet({
+	actionParameterName,
+	activeViewSettings,
+	apiURL,
+	appURL,
+	bulkActions,
+	creationMenu,
+	currentURL,
+	customDataRenderers,
+	customViewsEnabled,
+	filters,
+	formId,
+	formName,
+	id,
+	initialSelectedItemsValues,
+	inlineAddingSettings,
+	inlineEditingSettings,
+	items,
+	itemsActions,
+	namespace,
+	nestedItemsKey,
+	nestedItemsReferenceKey,
+	onActionDropdownItemClick,
+	onBulkActionItemClick,
+	overrideEmptyResultView,
+	pagination,
+	portletId,
+	selectedItemsKey,
+	selectionType,
+	showManagementBar,
+	showPagination,
+	showSearch,
+	sidePanelId,
+	sorting,
+	style,
+	views,
+}: IFrontendDataSetProps): JSX.Element;
+
+interface IFrontendDataSetProps {
+	actionParameterName?: string;
+	activeViewSettings?: string;
+	apiURL?: string;
+	appURL?: string;
+	bulkActions?: any[];
+	creationMenu?: {
+		primaryItems?: any[];
+		secondaryItems?: any[];
+	};
+	currentURL?: string;
+	customDataRenderers?: any;
+	customViewsEnabled?: boolean;
+	enableInlineAddModeSetting?: {
+		defaultBodyContent?: object;
+	};
+	filters?: (
+		| IAutocompleteFilter
+		| ICustomFilter
+		| ICheckboxFilter
+		| IDateRangeFilter
+		| IRadioFilter
+	)[];
+	formId?: string;
+	formName?: string;
+	id: string;
+	initialSelectedItemsValues?: any[];
+	inlineAddingSettings?: {
+		apiURL: string;
+		defaultBodyContent: object;
+	};
+	inlineEditingSettings?: [
+		boolean,
+		{
+			alwaysOn: boolean;
+			defaultBodyContent: object;
+		}
+	];
+	items?: any[];
+	itemsActions?: TItemsActions[];
+	namespace?: string;
+	nestedItemsKey?: string;
+	nestedItemsReferenceKey?: string;
+	onActionDropdownItemClick?: any;
+	onBulkActionItemClick?: any;
+	overrideEmptyResultView?: boolean;
+	pagination?: {
+		deltas?: [
+			{
+				href?: string;
+				label: number;
+			}
+		];
+		initialDelta: number;
+	};
+	portletId?: string;
+	selectedItemsKey?: string;
+	selectionType?: ['single', 'multiple'];
+	showManagementBar?: boolean;
+	showPagination?: boolean;
+	showSearch?: boolean;
+	sidePanelId?: string;
+	sorting?: [
+		{
+			direction?: ['asc', 'desc'];
+			key?: string;
+		}
+	];
+	style?: ['default', 'fluid', 'stacked'];
+	views: [
+		{
+			component?: any;
+			contentRenderer?: string;
+			contentRendererModuleURL?: string;
+			label?: string;
+			name?: string;
+			schema?: object;
+			thumbnail?: string;
+		}
+	];
+}
+
+type TItemsActions = {
+	data?: {
+		confirmationMessage?: string;
+		id?: string;
+		method?: 'delete' | 'get';
+		permissionKey?: string;
+	};
+	href?: string;
+	icon?: string;
+	label?: string;
+	target?: 'async' | 'headless' | 'link' | 'modal' | 'sidePanel';
+};
```

Comment:
There seems to be an error here. When I run gradle deploy on `notification-web`, I get:
```
src/main/resources/META-INF/resources/js/components/DefinitionOfTerms.tsx(185,7): error TS2322: Type '{ id: string; items: Item[]; itemsActions: { href: string; id: string; label: string; target: string; }[]; namespace: string; onActionDropdownItemClick: <T>({ action, itemData, }: { action: FDSAction; itemData: T; }) => void; ... 6 more ...; views: { ...; }[]; }' is not assignable to type 'IFrontendDataSetProps'.
  Types of property 'itemsActions' are incompatible.
    Type '{ href: string; id: string; label: string; target: string; }[]' is not assignable to type 'TItemsActions[]'.
      Type '{ href: string; id: string; label: string; target: string; }' is not assignable to type 'TItemsActions'.
        Types of property 'target' are incompatible.
          Type 'string' is not assignable to type '"async" | "headless" | "link" | "modal" | "sidePanel" | undefined'.
```

Reply:
Weird, I thought we can see some failures like this during a new `ant all` on CI or when `ant format-source-all` which is used for `ci:test:sf`